client/ayon_houdini/api/plugin.py

```diff
@@ -354,3 +355,36 @@ class HoudiniExtractorPlugin(publish.Extractor):
 
     hosts = ["houdini"]
     settings_category = SETTINGS_CATEGORY
+
+    def render_rop(self, instance: pyblish.api.Instance):
+        """Render the ROP node of the instance.
+
+        If `instance.data["frames_to_fix"]` is set and is not empty it will
+        be interpreted as a set of frames that will be rendered instead of the
+        full rop nodes frame range.
+
+        Only `instance.data["instance_node"]` is required.
+        """
+        # Log the start of the render
+        rop_node = hou.node(instance.data["instance_node"])
+        self.log.debug(f"Rendering {rop_node.path()}")
+
+        frames_to_fix = clique.parse(instance.data.get("frames_to_fix", ""),
+                                     "{ranges}")
+        if len(set(frames_to_fix)) > 1:
+            # Render only frames to fix
+            for frame_range in frames_to_fix.separate():
+                frame_range = list(frame_range)
+                self.log.debug(
+                    "Rendering frames to fix [{f1}, {f2}]".format(
+                        f1=frame_range[0],
+                        f2=frame_range[-1]
+                    )
+                )
+                # for step to be 1 since clique doesn't support steps.
+                frame_range = (
+                    int(frame_range[0]), int(frame_range[-1]), 1
+                )
+                render_rop(rop_node, frame_range=frame_range)
+        else:
+            render_rop(rop_node)
```

Comment:
Why is everybody making huge if statements?
```suggestion
        if len(set(frames_to_fix)) < 2:
            render_rop(rop_node)
            return

        # Render only frames to fix
        for frame_range in frames_to_fix.separate():
            frame_range = list(frame_range)
            frame_range.sort()
            first_frame = int(frame_range[0])
            last_frame = int(frame_range[-1])
            self.log.debug(
                f"Rendering frames to fix [{first_frame}, {last_frame}]"
            )
            # for step to be 1 since clique doesn't support steps.
            frame_range = (first_frame, last_frame, 1)
            render_rop(rop_node, frame_range=frame_range)
```

Reply:
I understand the thinking - but this to me really feels less clearly that there are two options and it's one of the two.

Also, the sort is not needed because `clique.Collection` when put into `list` should be sorted by design. Right?